### ISSUE TITLE
Setting screen as render function creates "trapdoor"

### ISSUE THREAD

**Issue by erksch** [react-navigation/react-navigation#4881]
# TL;DR

You can easily reproduce this by opening the last snack from the [authentication flows documentation](https://reactnavigation.org/docs/en/auth-flow.html) page: 
https://snack.expo.io/@react-navigation/auth-flow-v2 
Replace line 115 with `App: () => <AppStack />` and signing out will not work anymore.

### Current behavior

I created a SwitchNavigator with authentication flow and deep linking like suggested in the documentation and in [this article](https://medium.com/@ivanzotov/react-native-deep-linking-with-authorization-screen-drawer-and-nested-navigators-in-react-4f702b1520a0):

```javascript
export default createSwitchNavigator({
  Preloader,
  App: { screen: () => <AppStack uriPrefix="..." /> },
  Auth: AuthStack,
}, {
  initialRouteName: 'Preloader',
});
```

The `Preloader` decides whether the user is already authenticated or not and navigates either to `Auth` or `App`. If the user is not authenticated he can sign in via a screen in the `AuthStack` and navigates to the `AppStack` where deep links are handled. So far so good and everything works fine: navigating from `Auth` to `App` works. 

But I have a SignOut screen in the `AppStack` where the user is navigated back to the `AuthStack` after signing out. This part fails. Nothing happens, so I assume the `AuthStack` was not found (I tried navigating to `SignIn`, `Auth` or any other route outside of `AppStack` but nothing worked). 
When I remove the render function `() => <AppStack uriPrefix="..." />` and do: 

```
{
  Preloader,
  App: AppStack,
  Auth: AuthStack,
}
```

everything works fine. But obviously that's not what I want because I can't handle deep links anymore.
Using a render function for the screen property seems to always cause the stack to be "trapped".

### Expected Behavior

Even when using a render function for my stack to apply the uri prefix navigating to other stacks should work.

### How to reproduce

Create a SwitchNavigator like above and apply a render function to one of the routes. 

### My environment

| software         | version
| ---------------- | -------
| react-navigation | 2.11.2 |
| react-native     | 0.55 |
| node             | 8.11.3
| npm or yarn      | 1.7.0 |

**Comment by erksch** [comment]
I would use a workaround like setting the `uriPrefix` to my `SwitchNavigator` and not to `AppStack` but it does not work either: the nested deep links in `AppStack` are not triggered. 

If the problem is the render function `() =>` maybe there could be an NavigatorConfig parameter called `uriPrefix` so I can set the prefix via the config like this:
```javascript 
const AppStack = createStackNavigator({ ... }, { uriPrefix: '...' })
```

Is there a reason that this is not possible?

**Comment by erksch** [comment]
I created a [fork](https://github.com/erksch/react-navigation) with the described functionality.
This is the additional commit: https://github.com/erksch/react-navigation/commit/2df728f371374fe65647c4f30da60f501c44648f

Basically I added a config parameter to the `createNavigationContainer` function to pass the config of `createContainedStackNavigator`. 

`react-navigation-stack` has to be edited to in order to call `createNavigationContainer` properly with the config parameter. But sadly the repo has to builded. But I have added the parameter in https://github.com/erksch/react-navigation-stack/commit/63dab0c45a1e17e747ade5c0f907a8f90ec22e3e.

I would be glad if you'd take a look at my changes.

**Comment by brandonarbini** [comment]
This is my exact problem. I haven't had a chance to look at the solution/fork, but I was really surprised to not see `uriPrefix` as a config option and _only_ a prop on a navigator component. Without this I don't see how the recommended nested navigator pattern works with deep linking at all.

**Comment by brentvatne** [comment]
@erksch - the problem you're seeing here is due to this: https://reactnavigation.org/docs/en/common-mistakes.html#explicitly-rendering-more-than-one-navigator

a solution is mentioned there

**Comment by brandonarbini** [comment]
Thanks @brentvatne. But, according to https://reactnavigation.org/docs/en/auth-flow.html#set-up-our-navigators, how would you deep link into the app navigator stack? When adding the `uriPrefix` to the root stack it won't recognize deep links in the nested navigators (at least in my experience so far).

**Comment by brentvatne** [comment]
to be honest I'm not very familiar with the deep linking side of react-navigation, @ericvicenti would need to help with that

**Comment by InVincible2016** [comment]
Try:
```js
export default createSwitchNavigator({
  Preloader,
  App: { screen: AppStack, uriPrefix: "..." },
  Auth: AuthStack,
}, {
  initialRouteName: 'Preloader',
});
```

**Comment by erksch** [comment]
@InVincible2016 Amazing! This seems to work. Thank you very much.
